src/mappings/util.ts

```diff
@@ -138,28 +149,92 @@ export const getSigner = (extrinsic: SubstrateExtrinsic): string => {
   );
 };
 
-export const harvesterLikeParamsToObj = (
-  params: HarvesterLikeCallArgs,
-  formatKey = true
-): Record<string, any> => {
-  const obj: Record<string, any> = {};
-  params.forEach(p => {
-    obj[formatKey ? snakeToCamelCase(p.name) : p.name] =
-      p.name === 'asset_type' ? Object.keys(p.value)[0] : p.value;
-  });
-  return obj;
+/**
+ * Parses a raw Asset Document
+ */
+export const getDocValue = (
+  doc: Codec
+): Pick<AssetDocument, 'name' | 'link' | 'contentHash' | 'type' | 'filedAt'> => {
+  const {
+    uri: link,
+    content_hash: documentHash,
+    name,
+    doc_type: type,
+    filing_date: filedAt,
+  } = JSON.parse(doc.toString());
+
+  const hashType = Object.keys(documentHash)[0];
+
+  return {
+    name,
+    link,
+    contentHash: documentHash[hashType],
+    type,
+    filedAt,
+  };
 };
 
-export const formatAssetIdentifiers = (
-  identifiers: Record<string, string>[]
-): SecurityIdentifier[] =>
-  identifiers.map(i => {
+export const getSecurityIdentifiers = (item: Codec): SecurityIdentifier[] => {
+  const identifiers = JSON.parse(item.toString());
+  return identifiers.map(i => {
     const type = Object.keys(i)[0];
     return {
       type,
       value: i[type],
     };
   });
+};
+
+/**
+ * Parses a Vec<AssetCompliance>
+ */
+export const getComplianceRulesValue = (
+  requirements: Codec
+): Pick<Compliance, 'complianceId' | 'data'>[] => {
+  return JSON.parse(JSON.stringify(requirements));
+};
+
+/**
+ * Parses AssetCompliance
+ */
+export const getComplianceValue = (
+  compliance: Codec
+): Pick<Compliance, 'complianceId' | 'data'> => {
+  const { id, ...data } = JSON.parse(compliance.toString());
+  return {
+    complianceId: Number(id),
+    data,
+  };
+};
+
+/**
+ * Parses AssetTransferManager
+ */
+export const getTransferManagerValue = (
+  manager: Codec
+): Pick<TransferManager, 'type' | 'value'> => {
+  const { countTransferManager, percentageTransferManager } = JSON.parse(JSON.stringify(manager));
+
+  if (countTransferManager) {
+    return {
+      type: TransferRestrictionType.Count,
+      value: Number(countTransferManager),
+    };
+  }
+
+  if (percentageTransferManager) {
+    return {
+      type: TransferRestrictionType.Percentage,
+      value: Number(countTransferManager),
```

Comment:
`percentageTransferManager` ?

Reply:
This is a sample of params received. 
for % -
`[{"value":"0x7b5de4a8dfb706a24f5288082c3e4efdd834309785f7bee95ecea0d8f3abd80a"},{"value":"UNDVIDED"},{"value":{"PercentageTransferManager":50000}}]`

for count - `[{"value":"0x2e03244bbd44db4320b0a992f7e8eda9c88931be6c2e828b86147228531cf280"},{"value":"UTP-TC-0001"},{"value":{"CountTransferManager":45000}}]`

Reply:
wouldn't countTransferManager be undefined here then? (also surprised at the PascalCase vs camelCase being OK)